test/specs/DraggingHandlesSpec.js

```diff
@@ -210,7 +219,30 @@ describe("Dragging handles tests", function() {
         // End with mouse up
         testSlider.mouseup(mouseRight);
         expect(testSlider.getValue()).toEqual([2, 5]);
+    });
+
+    it("Should trigger change on mouseup", function() {
+        var ups = 0;
+        var changes = 0;
+
+        testSlider.on('slideStop', function(){
+            expect(changes).toBe(1);
+            ups++;
+        });
 
+        testSlider.mousedown(mouseEvent('mousedown', tickOffsets[1]));
```

Comment:
You're right. Instead of calling the `mousedown()` or `mouseup()` handlers directly, you can use `dispatchEvent()`.

Possible changes:
```js
testSlider.on('change', function(){
    expect(ups).toBe(0);
    changes++;
    expect(testSlider.getValue()).toEqual([2, 5]);
    expect(changes).toBe(1);
    expect(ups).toBe(1);
    done();
}

// Dispatch to slider
testSlider.sliderElem.dispatchEvent(mouseEvent('mousedown', tickOffsets[1]));
// Should bubble up and be caught by `document`
testSlider.sliderElem.dispatchEvent(mouseEvent('mouseup', tickOffsets[1]));
```

Reply:
@jespirit i just made a PR to correct a lot of our event callback triggering...see this PR: https://github.com/seiyria/bootstrap-slider/pull/902